tbdex/rfq/rfq.go

```diff
@@ -0,0 +1,124 @@
+package rfq
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/TBD54566975/tbdex-go/tbdex"
+	"github.com/tbd54566975/web5-go/dids/did"
+)
+
+// RFQKind identifies this message kind
+const RFQKind = "rfq"
+
+// RFQ represents a request for quote message within the exchange.
+type RFQ struct {
+	MessageMetadata tbdex.MessageMetadata `json:"metadata"`
+	Data            RFQData               `json:"data"`
+	PrivateData     *RFQPrivateData       `json:"privateData,omitempty"`
+	Signature       string                `json:"signature"`
+}
+
+// RFQData encapsulates the data content of a request for quote.
+type RFQData struct {
+	OfferingID string               `json:"offeringId"`
+	Payin      SelectedPayinMethod  `json:"payin"`
+	Payout     SelectedPayoutMethod `json:"payout"`
+	ClaimsHash string               `json:"claimsHash,omitempty"`
+}
+
+// RFQPrivateData contains data which can be detached from the payload without disrupting integrity.
+type RFQPrivateData struct {
+	Salt   string                 `json:"salt"`
+	Claims []string               `json:"claims"`
+	Payin  *PrivatePaymentDetails `json:"payin"`
+	Payout *PrivatePaymentDetails `json:"payout"`
```

Comment:
why are some of these (private data, payin, payout) pointers?

Reply:
They're optional/can be absent!